monkey/monkey_island/cc/services/config.py

```diff
@@ -538,12 +539,26 @@ def _format_icmp_scan_from_flat_config(config: Dict):
     @staticmethod
     def _format_fingerprinters_from_flat_config(config: Dict):
         flat_fingerprinter_classes_field = "finger_classes"
+        flat_http_ports_field = "HTTP_PORTS"
 
-        formatted_fingerprinters = config[flat_fingerprinter_classes_field]
-        config.pop(flat_fingerprinter_classes_field)
+        formatted_fingerprinters = [
+            {"name": f, "options": {}} for f in sorted(config[flat_fingerprinter_classes_field])
+        ]
+
+        if "HTTPFinger" in config[flat_fingerprinter_classes_field]:
+            for fp in formatted_fingerprinters:
+                if fp["name"] == "HTTPFinger":
+                    fp["options"] = {"http_ports": sorted(config[flat_http_ports_field])}
 
+                fp["name"] = ConfigService._translate_fingerprinter_name(fp["name"])
+
+        config.pop(flat_fingerprinter_classes_field)
         return formatted_fingerprinters
 
+    @staticmethod
+    def _translate_fingerprinter_name(name: str):
```

Comment:
Whats the purpose of this translation? To translate fingerprinter name to service name?

Reply:
Missing returned type

Reply:
To remove names like `HTTPFinger` or `MSSQLFinger` from the config. They represent an unnecessary coupling between the island and the agent. In this case, "http" is the name of the "plugin". When the "plugins" become real plugins, they will declare their own names and none of this will be necessary.

Reply:
I don't think I understand. I think it's still coupled, if we change the name on config to `HTTPFing` this translation will break. Either way, we should probably add a comment here explaining why we do this translation, because for the current state of code it's unneeded and the reader doesn't know what you anticipate.

Reply:
Comment added.